### PR TITLE
indexer: Add new indexing functions

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

**Comment by mslipper** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3665** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3665" target="_blank">Graphite</a>  👈
    * **PR #3666** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3666" target="_blank">Graphite</a> 
      * **PR #3667** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3667" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3665?utm_source=stack-comment).

File: indexer/services/query/erc20.go
Comment (tynes) [inline indexer/services/query/erc20.go:16]: This is fine but makes me think we should vendor in multicall3 so we can pack multiple eth calls into a single request

File: indexer/services/addresses.go
Comment (tynes) [inline indexer/services/addresses.go:17]: I think you should add compile time checks that `LegacyAddresses` and `BedrockAddresses` implement `AddressManager`

File: indexer/services/addresses.go
Comment (tynes) [inline indexer/services/addresses.go:44]: There is no getter for the `L2StandardBridge`?

File: indexer/services/l1/bridge/portal.go
Comment (tynes) [inline indexer/services/l1/bridge/portal.go:18]: Do you need to pass in the bindings here? Can't you just pass in the address and client and then create the bindings inside of this? Because creating the bindings needs the address, it feels like a strange api

File: indexer/services/query/erc20.go
Comment (tynes) [inline indexer/services/query/erc20.go:11]: Should this be named `NewERC20`?

File: indexer/services/l1/bridge/portal.go
Comment (tynes) [reply indexer/services/l1/bridge/portal.go:18]: The erc20 constructor follows the pattern noted above

File: indexer/services/l1/bridge/bridge.go
Comment (tynes) [inline indexer/services/l1/bridge/bridge.go:17]: Perhaps not a bad idea to add some docstrings here

**Comment by tynes** [comment]
@mslipper all other PRs LGTM, this is the only one that still has open comments

File: indexer/services/addresses.go
Comment (mslipper) [reply indexer/services/addresses.go:44]: Correct, since we know the address in advance.

Comment (mslipper) [reply indexer/services/addresses.go:46]: Ah now I understand what you mean - this is fixed.

**Review by tynes — APPROVED** [review]

**Review by smartcontracts — APPROVED** [review]

**Comment by mslipper** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/3665?utm_source=stack-comment).


Job Reference: bRbvLgKH9rDPN30uMQJy

**Pull request merged**
